components/cam/src/physics/cam/nudging.F90

```diff
@@ -1391,10 +1391,7 @@ subroutine nudging_timestep_init(phys_state)
        Nudge_ON=.true.
      else
        Nudge_ON=.false.
-       if(masterproc) then
-         write(iulog,*) 'NUDGING: WARNING - analyses file NOT FOUND. Switching '
-         write(iulog,*) 'NUDGING:           nudging OFF to coast thru the gap. '
-       endif
+       call endrun('NUDGING: Nudging data file NOT FOUND')
```

Comment:
I have a minor suggestion just for improving the error message. Can we do something like:
```
use shr_log_mod ,     only: errMsg => shr_log_errMsg
...

character(len=cxx)  :: err_str
....
....

write(err_str,*) 'NUDGING: Nudging data file(',trim(adjustl(Nudge_File)),')  NOT FOUND', errmsg(__FILE__, __LINE__)
call endrun(err_str)
```

You can modify the error message the way you like but I wanted to include the filename so that user knows which file is missing. The error message will also print the F90 file name and the line number where the error occurred.

Reply:
@singhbalwinder:  I will make adjustments following your suggestion and push the changes. Thank you!

Reply:
Shixuan, I remember the filename will be printed out in nudging_update_analyses_se.

Reply:
Hi @kaizhangpnl: I checked the nudging.f90 file again. I think that in Line 1342-1344: 
     if(masterproc) then
      write(iulog,*) 'NUDGING: Reading analyses:',trim(Nudge_Path)//trim(Nudge_File)
     endif
     
These codes are executed before the call of nudging_update_analyses_se(), but I think that you are right that the filename is actually printed out. 

Hi @singhbalwinder: I think that @kaizhangpnl raised a good point. The nudging file name has already been printed out before the endrun.

Reply:
Sounds good! Thanks @zhangshixuan1987 !